src/gui/qgsextentgroupbox.h

```diff
@@ -199,6 +218,10 @@ class GUI_EXPORT QgsExtentGroupBox : public QgsCollapsibleGroupBox, private Ui::
     QPointer< const QgsMapLayer > mExtentLayer;
     QString mExtentLayerName;
 
+    QgsMapToolExtent *mMapToolExtent = nullptr;
```

Comment:
This should be a unique_ptr i think. At the moment it's parented to the canvas and not deleted in the dialog - that means it hangs around forever. Safer to make it a unique_ptr so it's deleted with the dialog, and removing any risk of those annoying ghost rubber bands hanging around if something goes funny.